### PR TITLE
feat: split apk files

### LINKED ISSUE #104: feat: `ApkPatch`
### Type

Functionality

### Issue

Afaik, there are only 2 patch(BytecodePatch and ResourcePatch) at the moment. And ResourcePatch will not work if there is a resource decode problem with apktool. So with ApkPatch(or other name suitable) can at least help add raw resources to `assets` in some case.

### Feature

It is a patch can edit the `apk` by add, remove or edit the file in `apk` like file in `assets`, `lib`, ...

### Motivation

I see some people complain about the `universal apk` size. So which this `ApkPatch` it help to add a patch to reduce the `apk` size by remove unnecessary file in `apk` like not target `shared library`. In addition, it can also help add raw resources to `assets` if there is a patch need it and the apk's resources can not be decoded.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by oSumAtrIX**
## About

- This PR adds support for split apks. The API usage can be seen on the implementing CLI branch: https://github.com/revanced/revanced-cli/tree/feat/split

- Closes #104, #73.

## TODO

- [ ] Test the PR
- [ ] Implement TODOs

### CODE REVIEW

File: src/test/kotlin/app/revanced/patcher/usage/resource/patch/ExampleResourcePatch.kt [hunk 1]
```diff
@@ -18,6 +18,7 @@ import org.w3c.dom.Element
 @Version("0.0.1")
 class ExampleResourcePatch : ResourcePatch() {
     override fun execute(data: ResourceData): PatchResult {
+        data
```
Comment (Sculas) [inline src/test/kotlin/app/revanced/patcher/usage/resource/patch/ExampleResourcePatch.kt:21]:
```suggestion
```

**Review by Sculas — CHANGES_REQUESTED** [review]

**Comment by oSumAtrIX** [comment]
@Sculas the problem with just generically detecting whether it is a split or not is, that after patching it is required to know which apk is what, the language, assets or other apks. Now one might say, why not supply one apk each to the patcher, the problem with that is, that a patch would not be able to access all apks at the same time, like it does currently with a full apk, which is why I went with a sealed Apk class and a defined set of Subclasses for the splits.

**Comment by Sculas** [comment]
I understand, but all Java code is only ever in the base APK. All other splits do not contain dex files. So the problem you're trying to solve never existed to begin with.

**Comment by Sculas** [comment]
Either way, if this implementation works then it's good enough. Improvements can be made with time.

**Review by bao000 — COMMENTED** [review]
<dependency>
  <groupId>app.revanced</groupId>
  <artifactId>revanced-patcher</artifactId>
  <version>6.3.0</version>
</dependency>

**Comment by oSumAtrIX** [comment]
Superseded by #165.

**Pull request closed without merging**